github-runner-provisioner/requesthandler.go

```diff
@@ -61,10 +61,20 @@ func handleProvisioningRequest(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	log.Printf("Job in %s repo requested a %s runner\n", *workflowJobEvent.Repo.Name, jobLabel)
+
+	runnerLabels := []string{0: jobLabel}
+	if isRunnerAvailable(r.Context(), *workflowJobEvent.Repo.Owner.Login, *workflowJobEvent.Repo.Name, runnerLabels) {
+		log.Printf("%s runner already available. No action scaling action required.", jobLabel)
+		if _, err := w.Write([]byte("OK")); err != nil {
+			log.Printf("Error sending HTTP response: %v", err)
+		}
+	}
+
 	dryRun := len(r.Form["dry-run"]) > 0 && r.Form["dry-run"][0] == "true"
 	if err := runnerFunction(r.Context(), *workflowJobEvent.Repo.Owner.Login, *workflowJobEvent.Repo.Name, dryRun); err != nil {
-		log.Printf("Error creating Mac M1 runner for job %s [%s]: %v", *workflowJobEvent.WorkflowJob.Name, *workflowJobEvent.WorkflowJob.HTMLURL, err)
-		http.Error(w, fmt.Sprintf("Error creating Mac M1 runner: %v", err), http.StatusBadRequest)
+		log.Printf("Error creating %s runner for job %s [%s]: %v", jobLabel, *workflowJobEvent.WorkflowJob.Name, *workflowJobEvent.WorkflowJob.HTMLURL, err)
+		http.Error(w, fmt.Sprintf("Error creating %s runner: %v", jobLabel, err), http.StatusBadRequest)
```

Comment:
I know you didn't change this, but I think that an error at this point is not caused by the client, so `http.StatusBadRequest` doesn't make sense.

Maybe a 500 error would be better?